src/foam/dao/AbstractJDAO.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * @license
+ * Copyright 2017,2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao;
+
+import foam.core.*;
+import foam.lib.json.ExprParser;
+import foam.lib.json.JSONParser;
+import foam.lib.json.Outputter;
+import foam.lib.json.OutputterMode;
+import foam.lib.parse.*;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+import foam.nanos.auth.User;
+import foam.util.SafetyUtil;
+import java.io.*;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.regex.Pattern;
+import java.util.TimeZone;
+
+public abstract class AbstractJDAO
+    extends ProxyDAO
+{
+  protected Pattern COMMENT = Pattern.compile("(/\\*([^*]|[\\r\\n]|(\\*+([^*/]|[\\r\\n])))*\\*+/)|(//.*)");
```

Comment:
This seems like overkill.  Isn't it enough to just check if the first two characters are //?
Not really related to this PR though.

Reply:
Kind of is yeah but I wanted it to be exhaustive